subsys/fs/littlefs_fs.c

```diff
@@ -181,13 +181,50 @@ static void release_file_data(struct fs_file_t *fp)
 	fp->filep = NULL;
 }
 
-static int littlefs_open(struct fs_file_t *fp, const char *path)
+static int lfs_flags_from_zephyr(int zflags)
+{
+	int flags = 0;
+
+	if (zflags & FO_CREATE) {
+		flags |= LFS_O_CREAT;
+		zflags ^= FO_CREATE;
```

Comment:
What's with `^=`?  `&= ~` would be less startling (standard).

Reply:
Done.